Interop/Runtime/src/native/kotlin/kotlinx/cinterop/NativeTypes.kt

```diff
@@ -56,6 +56,20 @@ external fun CPointer<*>.getRawValue(): NativePtr
 
 internal fun CPointer<*>.cPointerToString() = "CPointer(raw=$rawValue)"
 
+
+public final class NativeVector private constructor()
```

Comment:
This class should also have `equals`/`hashCode`/`toString` implemented (and tested).
`==` equality should be supported too, see https://github.com/JetBrains/kotlin-native/blob/e1b3fefd3333fa730bb00174b7eb24c90822f6a0/runtime/src/main/kotlin/kotlin/native/internal/Intrinsics.kt#L22